### PR TITLE
Render overlay to prompt user for a display name

### PR DESCRIPTION
**Pull request by ker-an**
Users require a display name in order to comment on an article.

### CODE REVIEW

File: src/js/stream.js [hunk 1]
```diff
@@ -20,12 +21,20 @@ class Stream {
...
+					this.renderDisplayNameOverlay();
+				}
+
 				if (this.token && this.embed) {
 					this.login();
 				}
 			});
 	}
 
+	static renderDisplayNameOverlay () {
```
Comment (ker-an) [inline src/js/stream.js]: This was done to make it easier to test that `displayNameOverlay()` is called and this way we don't have to rely on dispatching and listening for an event. I'm totally open to other solutions to this problem!

Comment (ker-an) [reply src/js/stream.js]: Also, because this function doesn't use `this`, according to Origami specs, it should be a static function.

Comment (ker-an) [reply src/js/stream.js]: Able to successfully stub `displayNameOverlay` so this function isn't needed any more!

File: src/js/stream.js [hunk 2]
```diff
@@ -31,21 +42,25 @@ class Stream {
 		}
 	}
 
-	getJsonWebToken () {
-		return fetch('https://comments-api.ft.com/user/auth/', {
+	getJsonWebToken (displayName) {
+		const url = displayName ?
```
Comment (tbergmen) [inline src/js/stream.js]: I think during the merging of branch `v6` in to this branch, this logic is lost:

https://github.com/Financial-Times/o-comments/blob/11318be3fa48c4dcd1841d10d30bcde1b5cc9e1b/src/js/stream.js#L35

Comment (ker-an) [reply src/js/stream.js]: I've re-added it!

Comment (tbergmen) [reply src/js/stream.js]: 👍

**Comment by GlynnPhillips** [comment]
Closing as this as has been replaced by https://github.com/Financial-Times/o-comments/pull/88

**Pull request closed without merging**